### PR TITLE
Add draw marker method to world class

### PR DESCRIPTION
**Pull request by zorg93**
Added an enum containing all available marker types aswell

### CODE REVIEW

**Comment by Nacorpio** [comment]
What is a marker? :cactus:

**Comment by zorg93** [comment]
things like these, ignore the rest of the menu 
image

**Comment by crosire** [comment]
To keep the same naming convention I'd call it "World::AddMarker".

**Comment by JohnnyCrazy** [comment]
BTW, do we need to call this every tick?

**Comment by zorg93** [comment]
Renamed it to DrawMarker, AddMarker makes it seem like you only need to call it once, DrawMarker(Like other Draw methods) must be called on every tick you want them to show

**Comment by Frede1012** [comment]
Should we create a wrapper that draws this ontick  automatically or let the users handle it themselves ?

**Comment by Nacorpio** [comment]
I'm sorry for asking too many questions, but what are rays?

**Comment by zorg93** [comment]
@Frede1012 I thought about doing it like that, but that is a much more involved method and there isnt any specific gain from it
@Nacorpio http://docs.unity3d.com/ScriptReference/Physics.Raycast.html theres so info on them in here. they are used in pretty much every 3d game engine

**Comment by Nacorpio** [comment]
That's pretty useful.

**Comment by crosire** [comment]
Alright, didn't know you have to call that every frame, "Draw" makes more sense then indeed.

**Comment by IncoCode** [comment]
How about ```Colour```? :smile:

**Comment by crosire** [comment]
@IncoCode ieks!

**Comment by IncoCode** [comment]
@crosire then it should be changed: https://github.com/crosire/scripthookvdotnet/pull/161/files#diff-70aad3c61652616ed1bee696a442670fR374 :smirk:

**Comment by crosire** [comment]
Right, although it's only a parameter name.

**Comment by zorg93** [comment]
I hope your not complaining about the correct spelling of the word colour

**Comment by crosire** [comment]
There is no "correct" spelling, but since .NET throughout uses the "o" instead of "ou" notation it makes the most sense to follow that here too.

**Comment by zorg93** [comment]
@crosire is that better?

File: source/World.hpp [hunk 1]
```diff
@@ -67,6 +67,38 @@ namespace GTA
 		SmallExplosion7 = 33,
 		Explosion6 = 34
 	};
+	public enum class MarkerTypes
+	{
+		UpsideDownCone = 0,
+		VerticalCylinder = 1,
+		Thick_Chevron_Up = 2,
```
Comment (JohnnyCrazy) [inline source/World.hpp]: Going with the naming convention, could you change all the names to the CamelCase-style? Like this to `ThickChevronUp`. :)

File: source/World.cpp [hunk 2]
```diff
@@ -367,4 +367,22 @@ namespace GTA
 	{
 		return gcnew RayCastResult(Native::Function::Call<int>(Native::Hash::_0x377906D8A31E5586, source.X, source.Y, source.Z, target.X, target.Y, target.Z, static_cast<int>(options), entity == nullptr ? 0 : entity->Handle, UnkFlags));
 	}
+	void World::DrawMarker(MarkerTypes type, Vector3 pos, Vector3 dir, Vector3 rot, Vector3 scale, System::Drawing::Color Color)
+	{
+		DrawMarker(type, pos, dir, rot, scale, Color, false, false, 2, false, nullptr, nullptr, false);
+	}
+	void World::DrawMarker(MarkerTypes type, Vector3 pos, Vector3 dir, Vector3 rot, Vector3 scale, System::Drawing::Color Color, bool BobUpAndDown, bool FaceCam_Y, int UNK_2, bool Rotate_Y, System::String ^TextueDict, System::String ^TextureName, bool DrawOnEnt)
```
Comment (JohnnyCrazy) [inline source/World.cpp]: Also some naming-differences. Last parameters should be lower-camelcase too.

**Comment by JohnnyCrazy** [comment]
Left some notes, after this I will happily merge this :+1: 

(If you don't have time fixing it, just say it, then I will do it after the merge)

**Comment by zorg93** [comment]
@JohnnyCrazy this the style you want?

**Comment by JohnnyCrazy** [comment]
Perfect, thanks! :+1:

**Comment by HappyCrayfish** [comment]
Look in the drawmarker's last param.

void OnTick(object sender, EventArgs e)
    {
        if (startupMessagesDisplayed == false)
        {
            startupMessagesDisplayed = true;
            UI.ShowSubtitle("Mod Enabled");
        }

        if (onMission == true)
        {
            World.DrawMarker(MarkerTypes.VerticleCircle, new Vector3(237, 779, 28163), Vector3.WorldUp, Vector3.Zero, new Vector3(10, 10, 10), /*what to put here to get the RightKey color?*/);
        }

        UI.ShowSubtitle("Player Vector3 : " +player.Position.X +player.Position.Y +player.Position.Z);
    }

**Pull request merged**